e2eTests/scenarios/BCApps/runtest.ps1

```diff
@@ -0,0 +1,126 @@
+
+Remove-Module e2eTestHelper -ErrorAction SilentlyContinue
+Import-Module (Join-Path $PSScriptRoot "..\..\e2eTestHelper.psm1") -DisableNameChecking
+
+$branch = "main"
+$template = "https://github.com/$pteTemplate"
+
+$sourceRepo = "microsoft/BCApps"
+
+$repoName = [System.IO.Path]::GetFileNameWithoutExtension([System.IO.Path]::GetTempFileName())
+Push-Location
+$repository = "$githubOwner/$repoName"
+
+# Login
+SetTokenAndRepository -github:$github -githubOwner $githubOwner -token $e2epat -repository $repository
+
+# Create repository1
+CreateAlGoRepository `
+    -github:$github `
+    -linux:$linux `
+    -template "https://github.com/$sourceRepo" `
+    -repository $repository `
+    -branch $branch
+
+$repoPath = (Get-Location).Path
+Write-Host "Repo Path: $repoPath"
+
+RunUpdateAlGoSystemFiles -directCommit -wait -templateUrl $template -ghTokenWorkflow $e2epat -repository $repository | Out-Null
+
+SetRepositorySecret -repository $repository -name 'GHTOKENWORKFLOW' -value $algoauthapp
+
+CancelAllWorkflows -repository $repository
+
+# Pull and test workflows
+Pull
+
+$run = RunCICD -repository $repository -branch $branch -wait
+
+# Test that app artifacts are generated
+Test-ArtifactsFromRun -runid $run.id -folder '.artifacts' -repoVersion '*.*' -appVersion '*.*'
+$appCount = @(Get-ChildItem -Path '.artifacts/*-Apps-*/*.app'  -Recurse).Count
+$testAppCount = @(Get-ChildItem -Path '.artifacts/*-TestApps-*/*.app'  -Recurse).Count
+$appCount | Should -BeGreaterThan 0
+$testAppCount | Should -BeGreaterThan 0
+
+# Modify a file in the repository and create a Pull Request
+$branch = "e2etest"
+$title = "End 2 end test"
+invoke-git checkout -b $branch
+
+$fileToChange = Join-Path $repoPath "src/Tools/Performance Toolkit/App/src/BCPTTestSuite.Codeunit.al"
+$fileContent = Get-Content -Path $fileToChange -Encoding UTF8
+$fileContent[0] = $fileContent[0] + "// $title"
+Set-Content -Path $fileToChange -Value $fileContent -Encoding UTF8
+
+invoke-git add $fileToChange
+invoke-git commit -m $title
+invoke-git push --set-upstream origin $branch
+
+invoke-gh pr create --fill --head $branch --repo $repository --base main --body $title
+
+Start-Sleep -Seconds 60
+
+$prs = @(invoke-gh -returnValue pr list --repo $repository | Where-Object { $_.Contains($title) })
+if ($prs.Count -eq 0) {
+    throw "No Pull Request was created"
+}
+elseif ($prs.Count -gt 1) {
+    throw "More than one Pull Request exists"
+}
+
+$headers = GetHeaders -token $e2epat -repository "$($githubOwner)/.github"
+$url = "https://api.github.com/repos/$repository/actions/runs"
+$run = ((InvokeWebRequest -Method Get -Headers $headers -Uri $url -retry).Content | ConvertFrom-Json).workflow_runs | Where-Object { $_.event -eq 'pull_request' } | Where-Object { $_.name -eq 'Pull Request Build' }
+if (-not $run) {
+    throw "No Pull Request Build workflow run was found"
+}
+WaitWorkflow -repository $repository -runid $run.id
+
+# There should be only 1 app rebuilt
```

Comment:
Won't this depend on the settings we have in BCApps or do you update the settings somewhere? I guess this wouldn't be the expected outcome if we moved to using modifiedProjects instead for example.

Reply:
You are right. The intention of this end 2 end is to test BCApps - meaning that if we change BCApps, we should change the test as well - to maybe count the number of projects rebuilt instead of the apps.
It is also depending on one specific file in the repo, which might be removed or moved at some point in time.
In that case, we will also have to change the test.
I just want a test, which tests how BCApps works - and if we ever break BCApps again, we should add stuff to this test to avoid that that happens.

By modifying the test to test a specific mode, we are not testing BCApps.

Reply:
> meaning that if we change BCApps, we should change the test as well 

This sounds like something we'll definitely forget. 
In the case of this failure, we saw builds on BCApps fail. Testing that the build works should be sufficient IMO. Other E2E/Pester tests should cover testing the actual functionality and whether it works as intended. 

If this E2E test fails in the future I'd suggest we change this. For now, let's start by getting the fix in.